### PR TITLE
openssl: add an explicit BR on `perl{(Test::Harness), (Math::BigInt)}` to enable ptest

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Fix ptest build. All tests pass!

###### Change Log  

- Add an explicit BR on `perl{(Test::Harness), (Math::BigInt)}` to enable ptest

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Pipeline build id: Local Build with RUN_CHECK=y

### CODE REVIEW

File: SPECS/openssl/openssl.spec [hunk 1]
```diff
@@ -42,17 +39,22 @@ Patch19:        openssl-1.1.1-sp80056arev3.patch
 Patch20:        openssl-1.1.1-jitterentropy.patch
 Patch21:        openssl-1.1.1-drbg-seed.patch
 Patch22:        openssl-1.1.1-fips-SymCrypt.patch
+# Don't depend on bash by default
+%define __requires_exclude ^/(bin|usr/bin).*$
+%define soversion 1.1
```
Comment (PawelWMS) [inline SPECS/openssl/openssl.spec:44]: Could you move these back to the top of the file? They are a lot easier to find there.:)

**Review by PawelWMS — APPROVED** [review]
One short request, otherwise looks good!

File: SPECS/openssl/openssl.spec (hunk 1 above)
Comment (rlmenge) [reply SPECS/openssl/openssl.spec:44]: I think this is a linter recommendation that all %defines are moved after patches. We should fix this check

**Review by rlmenge — APPROVED** [review]
😎 LGTM

File: SPECS/openssl/openssl.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/openssl/openssl.spec:44]: Sure moving it back up :)

**Pull request merged**